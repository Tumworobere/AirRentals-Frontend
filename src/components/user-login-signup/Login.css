.login-form {
  max-width: 360px;
  padding: 20px;
  margin: 60px auto;
}

.login-form label {
  display: block;
  margin: 30px auto;
}

.login-form span {
  display: block;
  margin-bottom: 6px;
}

.login-form input {
  padding: 8px 6px;
  font-size: 1em;
  color: #777;
  width: 100%;
}

.or {
  margin-top: 10px;
}

.btn {
  background: none;
  border: 2px solid #1f9751;
  padding: 6px 12px;
  border-radius: 4px;
  color: #1f9751;
  font-weight: bold;
  margin-top: 10px;
  cursor: pointer;
  font-size: 1em;
}

.btn:hover {
  background: #1f9751;
  color: #fff;
}
