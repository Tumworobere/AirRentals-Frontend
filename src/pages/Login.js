/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable no-unused-expressions */
/* eslint-disable no-undef */
import React, { useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { downTop, topDown } from '../animations';
import { login } from '../logic/api';

const Login = () => {
  const navigate = useNavigate();
  useEffect(() => {
    document.getElementById('spinnerbg') ? spinnerbg.classList.add('hidden') : null;
    topDown();
  }, []);
  sessionStorage.getItem('token') !== null ? navigate('/') : null;
  return (
    <div className="mt-0 w-screen">
      <div id="main" className="opacity-0 block p-6 rounded-lg shadow-lg bg-white w-3/4 md:w-1/2 lg:w-1/3 h-auto m-auto mt-48 mb-24 -translate-y-full">
        <h1 className="mt-3 text-center text-3xl font-bold text-gray-900">Log In </h1>
        <form id="form">
          <div className="form-group">
            <br />
            <input type="email" id="email" placeholder="Enter email address" required className="appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm m-0" />
          </div>
          <div className="form-group mb-6">
            <input id="password" type="password" placeholder="Enter password" required className="appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-b-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm m-0" />
          </div>
          <div className="flex justify-between items-center mb-6">
            <div className="form-group form-check text-center w-full flex justify-center">
              <input id="remember" type="checkbox" className="form-check-input appearance-none h-4 w-4 border border-gray-300 rounded-sm bg-white checked:bg-indigo-600 checked:border-indigo-600 focus:outline-none transition duration-200 mt-1 align-top bg-no-repeat bg-center bg-contain float-left mr-2 cursor-pointer" />
              <label htmlFor="remember" className="form-check-label inline-block text-gray-800 overf">Remember me</label>
            </div>
          </div>
          <button
            type="button"
            className="w-full
        px-6
        py-2.5
        text-white
        font-medium
        text-m
        leading-tight
        rounded-full
        bg-indigo-800
        shadow-md
        hover:bg-purple-700 hover:shadow-lg
        focus:bg-green-400 focus:shadow-lg focus:outline-none focus:ring-0
        active:bg-indigo-600 active:shadow-lg
        transition
        duration-150
        ease-in-out
        cursor-pointer"
            onClick={(e) => {
              e.preventDefault;
              const form = document.getElementById('form');
              login(form.email.value, form.password.value);
            }}
          >
            Log In
          </button>
          <p className="text-gray-800 mt-6 text-center">
            Not a member?
            <button
              type="button"
              onClick={() => {
                downTop();
                setTimeout(() => {
                  navigate('/register');
                }, 700);
              }}
              className="text-indigo-600 hover:text-indigo-700 focus:text-indigo-700 transition duration-200 ease-in-out"
            >
              {' '}
              Register
            </button>
            <br />
          </p>
        </form>
      </div>
    </div>
  );
};

export default Login;
