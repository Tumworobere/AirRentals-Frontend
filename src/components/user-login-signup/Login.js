/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable react/button-has-type */
import './Login.css';
import { useState } from 'react';
import { Link } from 'react-router-dom';

export default function Login() {
  // const [username, setUsername] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const handleSubmit = (event) => {
    event.preventDefault();
    // console.log(email,password);
  };

  return (
    <div className="h-screen w-screen">
      {/* <div className="opacity-0 block p-6 rounded-lg
      shadow-lg bg-white w-3/4 md:w-1/2 lg:w-1/3
      h-auto m-auto mt-48 transition-all duration-1000 -translate-y-full"> */}
      <h1 className="mt-6 text-center text-3xl font-extrabold text-gray-900">Log In </h1>
      <form className="login-form" onSubmit={handleSubmit}>
        <div className="form-group">
          <div className="form-group mb-6">
            <input
              type="email"
              className="appearance-none rounded-none relative
              block w-full px-3 py-2 border border-gray-300
               placeholder-gray-500 text-gray-900 rounded-t-md
               focus:outline-none focus:ring-indigo-500
               focus:border-indigo-500 focus:z-10 sm:text-sm"
              placeholder="enter email address"
              onChange={(e) => setEmail(e.target.value)}
              value={email}
            />
          </div>

          <div className="form-group mb-6">
            <input
              type="password"
              className="appearance-none rounded-none relative
               block w-full px-3 py-2 border border-gray-300
               placeholder-gray-500 text-gray-900 rounded-b-md
               focus:outline-none focus:ring-indigo-500 focus:border-indigo-500
               focus:z-10 sm:text-sm"
              placeholder="enter Password"
              onChange={(e) => setPassword(e.target.value)}
              value={password}
            />
          </div>

          <button className="
            w-full
      px-6
      py-2.5
      bg-indigo-800
      text-white
      font-medium
      text-m
      leading-tight
      rounded-full
      shadow-md
      hover:bg-purple-700 hover:shadow-lg
      focus:bg-purple-700 focus:shadow-lg focus:outline-none focus:ring-0
      active:bg-purple-800 active:shadow-lg
      transition
      duration-150
      ease-in-out
      cursor-pointer"
          >
            Login

          </button>

          <p className="text-gray-800 mt-6 text-center">
            Not a member?
            <button
              type="button"
              className="text-indigo-600 hover:text-indigo-700
focus:text-indigo-700 transition duration-200 ease-in-out"
            >
              {' '}
              <Link to="/signup">Register</Link>
            </button>
            <br />
          </p>
        </div>
      </form>
    </div>
  //  </div>
  );
}
