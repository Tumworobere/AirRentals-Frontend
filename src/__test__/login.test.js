import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Login from '../pages/Login';
import store from '../redux/configureStore';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

it('Login renders the UI as expected', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Login />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
