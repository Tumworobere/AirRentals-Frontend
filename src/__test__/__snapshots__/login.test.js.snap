// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders the UI as expected 1`] = `
<div
  className="mt-0 w-screen"
>
  <div
    className="opacity-0 block p-6 rounded-lg shadow-lg bg-white w-3/4 md:w-1/2 lg:w-1/3 h-auto m-auto mt-48 mb-24 -translate-y-full"
    id="main"
  >
    <h1
      className="mt-3 text-center text-3xl font-bold text-gray-900"
    >
      Log In 
    </h1>
    <form
      id="form"
    >
      <div
        className="form-group"
      >
        <br />
        <input
          className="appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-t-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm m-0"
          id="email"
          placeholder="Enter email address"
          required={true}
          type="email"
        />
      </div>
      <div
        className="form-group mb-6"
      >
        <input
          className="appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 rounded-b-md focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 focus:z-10 sm:text-sm m-0"
          id="password"
          placeholder="Enter password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="flex justify-between items-center mb-6"
      >
        <div
          className="form-group form-check text-center w-full flex justify-center"
        >
          <input
            className="form-check-input appearance-none h-4 w-4 border border-gray-300 rounded-sm bg-white checked:bg-indigo-600 checked:border-indigo-600 focus:outline-none transition duration-200 mt-1 align-top bg-no-repeat bg-center bg-contain float-left mr-2 cursor-pointer"
            id="remember"
            type="checkbox"
          />
          <label
            className="form-check-label inline-block text-gray-800 overf"
            htmlFor="remember"
          >
            Remember me
          </label>
        </div>
      </div>
      <button
        className="w-full px-6 py-2.5 text-white font-medium text-m leading-tight rounded-full bg-indigo-800 shadow-md hover:bg-purple-700 hover:shadow-lg focus:bg-green-400 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-indigo-600 active:shadow-lg transition duration-150 ease-in-out cursor-pointer"
        onClick={[Function]}
        type="button"
      >
        Log In
      </button>
      <p
        className="text-gray-800 mt-6 text-center"
      >
        Not a member?
        <button
          className="text-indigo-600 hover:text-indigo-700 focus:text-indigo-700 transition duration-200 ease-in-out"
          onClick={[Function]}
          type="button"
        >
           
          Register
        </button>
        <br />
      </p>
    </form>
  </div>
</div>
`;
